### PR TITLE
fix(linter): false positive in `iframe-missing-sandbox`

### LINKED ISSUE #6750: linter: oxc for vscode, document.createElement("iframe") - missing a sandbox attribute
### What version of Oxlint are you using?

0.10.1

### What command did you run?

oxc vscode extension

### What does your `.oxlint.json` config file look like?

As far as I know there is no .oxlint.json

### What happened?

I have this line in a .js file:
```javascript
const i = document.createElement("iframe");
```

I get this from oxc:

> An iframe element is missing a sandbox attribute
> help: Add a `sandbox` attribute to the `iframe` element.oxc[eslint-plugin-react(iframe-missing-sandbox)]> 
> (https://oxc.rs/docs/guide/usage/linter/rules/eslint-plugin-react/iframe-missing-sandbox)

I am not using React. And I have no idea about the plugin mentioned. (And the link above gives 404.)

### PR DESCRIPTION
**Pull request by shulaoda**
closes #6750 

According to `eslint-plugin-react`, it seems that we should also search for `// @jsx pragma` in the source code.

### CODE REVIEW

File: crates/oxc_linter/src/utils/react.rs [hunk 1]
```diff
@@ -12,16 +12,13 @@ use oxc_semantic::AstNode;
...
-        }
-        Expression::ComputedMemberExpression(member_expr) => {
-            member_expr.static_property_name().is_some_and(|name| name == "createElement")
-        }
         Expression::Identifier(ident) => ident.name == "createElement",
-        _ => false,
+        expr => expr.is_specific_member_access(
+            ctx.settings().react.pragma.as_ref().map_or("React", |pragma| pragma.as_str()),
+            "createElement",
+        ),
```
Comment (mysteryven) [inline crates/oxc_linter/src/utils/react.rs:21]: There are also some false positive:

```js
const createElement = document.createElement
createElement('iframe')

const React = myReact
React.createElement('iframe')
```


Can we port  [`isDestructuredFromPragmaImport`](https://github.com/jsx-eslint/eslint-plugin-react/blob/d1556a3d2f95f56cfb6e7c0a341af16742f64033/lib/util/isCreateElement.js#L28-L29) here? we can ignore the `require` part for this issue.

File: crates/oxc_linter/src/config/settings/react.rs [hunk 2]
```diff
@@ -8,6 +8,8 @@ use serde::{Deserialize, Serialize};
 #[derive(Debug, Clone, Deserialize, Default, Serialize, JsonSchema)]
 #[cfg_attr(test, derive(PartialEq))]
 pub struct ReactPluginSettings {
+    pub pragma: Option<CompactStr>,
```
Comment (mysteryven) [inline crates/oxc_linter/src/config/settings/react.rs:11]: The `pragma` are in `react.pragma`: https://github.com/jsx-eslint/eslint-plugin-react#configuration-legacy-eslintrc-

We can also set the default serde value to remove `Option`

File: apps/oxlint/fixtures/print_config/ban_rules/expect.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
       "rootDir": []
     },
     "react": {
+      "pragma": "React",
```
Comment (mysteryven) [inline apps/oxlint/fixtures/print_config/ban_rules/expect.json:26]: Why not keep same with ESLint config file? https://github.com/oxc-project/oxc/blob/3923e63b6100104a9985dec6fa27780471e88d78/crates/oxc_linter/src/config/oxlintrc.rs#L16

File: crates/oxc_linter/src/utils/react.rs (hunk 1 above)
Comment (mysteryven) [reply crates/oxc_linter/src/utils/react.rs]: Or don't add `ctx.settings().react.pragma` for this issue:

https://github.com/oxc-project/oxc/blob/6f2253886ddb461063acc45e7647c2805adc4d65/crates/oxc_linter/src/utils/react.rs#L147

File: apps/oxlint/fixtures/print_config/ban_rules/expect.json (hunk 3 above)
Comment (shulaoda) [reply apps/oxlint/fixtures/print_config/ban_rules/expect.json:26]: `form_components` and `link_components` don't seem to be in `react` but we designed them this way from the beginning.
```rust
// <https://github.com/jsx-eslint/eslint-plugin-react#configuration-legacy-eslintrc->
#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
#[cfg_attr(test, derive(PartialEq))]
pub struct ReactPluginSettings {
    #[serde(default = "default_pragma")]
    pub pragma: CompactStr,

    #[serde(default)]
    #[serde(rename = "formComponents")]
    form_components: Vec<CustomComponent>,

    #[serde(default)]
    #[serde(rename = "linkComponents")]
    link_components: Vec<CustomComponent>,
    // TODO: More properties should be added
}
```

Comment (shulaoda) [reply apps/oxlint/fixtures/print_config/ban_rules/expect.json:26]: https://github.com/jsx-eslint/eslint-plugin-react#configuration-legacy-eslintrc

**Comment by mysteryven** [comment]
Close in favor of #6872

File: apps/oxlint/fixtures/print_config/ban_rules/expect.json (hunk 3 above)
Comment (mysteryven) [reply apps/oxlint/fixtures/print_config/ban_rules/expect.json:26]: My fault, I missed this. Seems a simple way to fix this issue has merged.

**Pull request closed without merging**